crypto/evp/evp_pkey.c

```diff
@@ -85,7 +85,6 @@ EVP_PKEY *EVP_PKCS82PKEY_ex(const PKCS8_PRIV_KEY_INFO *p8, OSSL_LIB_CTX *libctx,
         /* try legacy */
         pkey = evp_pkcs82pkey_legacy(p8, libctx, propq);
 
- end:
     OPENSSL_clear_free(encoded_data, encoded_len);
     OSSL_DECODER_CTX_free(dctx);
```

Comment:
Maybe a NULL check inside the function is a good idea?

Reply:
Added FWIW.